### PR TITLE
Ruby branch

### PR DESCRIPTION
**Pull request by kedskeds**

### CODE REVIEW

**Comment by kedskeds** [comment]
@bontgoy Hey Tyler, could you please provide me with feedback on 4.4? Thank you.

**Comment by Bontgoy** [comment]
If requesting feedback make sure to wait for my reply before merging the pull request

File: ruby/vampires.rb [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	name = name.downcase
+	if(name == "drake cula") || (name == "tu fang")
+		valid_name = false
+	end
+
+	puts "How old are you? What year were you born?"
+	age = gets.chomp
+	birth_year = gets.chomp
+
+	if (2016 - birth_year.to_i) != age.to_i
```
Comment (Bontgoy) [inline ruby/vampires.rb:26]: with this setup you will have to change this file every year for it to work.

Check out the Ruby `Time` class. You should be able to handle a dynamic year check with something like `Time.now.year`

File: ruby/vampires.rb [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+	if valid_age && (eats_garlic || wants_ins)
+		result = "Probably not a vampire."
+	end
+	if (!valid_age && (!eats_garlic || !wants_ins)) || (likes_sunshine == false)
+		result = "Probably a vampire."
+	end
+	if !valid_age && !eats_garlic && !wants_ins
+		result = "Almost certainly a vampire."
+	end
+	if !valid_name
```
Comment (Bontgoy) [inline ruby/vampires.rb:65]: I think this should be your first check. If their name is Drac or Fang then we know we have a vampire. You can nest the rest of the checks under an else following this check. if vamp == true then no reason to check anything else.

The below is an example of handling the control flow in a more efficient computational pattern.  Also, it is important to practice nesting conditions as many control flows will require you to do so.

```
if name == vamp
      vampire = true
else
      if other conditions
      elsif other conditions
      end
end
```

**Pull request merged**